transforms/tabular/tabular-thresholding-plugin/src/func.py

```diff
@@ -0,0 +1,165 @@
+import json
+import logging
+import os
+import pathlib
+import re
+from typing import Optional
+import numpy as np
+import vaex
+from thresholding import custom_fpr
+from thresholding import n_sigma
+from thresholding import otsu
+import warnings
+
+logger = logging.getLogger("main")
+
+
+def thresholding_func(
+    csvfile: str,
+    inpDir: pathlib.Path,
+    outDir: pathlib.Path,
+    negControl: str,
+    posControl: str,
+    variableName: str,
+    thresholdType: str,
+    mappingvariableName: Optional[str],
+    metaDir: Optional[pathlib.Path],
+    falsePositiverate: Optional[float] = 0.1,
+    numBins: Optional[int] = 512,
+    n: Optional[int] = 4,
+    outFormat: Optional[str] = "csv",
+) -> None:
+
+    """Computes variable threshold using negative or negative and positive control data, and determines if the variable value of
+    each ROI is above or below threshold. The control data used for computing threshold depends on the type of thresholding methods
+    https://github.com/nishaq503/thresholding.git.
+    Args:
+        csvfile (str) : Filename
+        inpDir (Path) : Path to tabular data directory
+        outDir (Path) : Path to output directory
+        negControl (str):FeatureName containing information of non treated wells
+        posControl (str):FeatureName containing information of wells with the known treatment
+        variableName (str):FeatureName for computing thresholds
+        thresholdType (str):Name of threshold method
+        mappingvariableName (str) optional: Feature name to merge tabular data with metadata
+        metaDir (Path) optional : Path to metadata directory
+        falsePositiverate (float) optional, default 0.1: Tuning parameter
+        numBins (int) optional, default 512: Number of bins
+        n (int) optional, default 4: Number of standard deviation away from mean value
+        outFormat (str) optional, default csv: File format of an output file
```

Comment:
Remove the type hints from the doc-string. They are redundant when specified in the function signature.